wandb/apis/public.py

```diff
@@ -2168,8 +2328,8 @@ def _load(self):
             or response.get('project') is None \
                 or response['project'].get('artifact') is None:
             # we check for this after doing the call, since the backend supports raw digest lookups
-            # which don't include ":"
-            if ':' not in self.artifact_name:
+            # which don't include ":" and are 32 characters long
+            if ':' not in self.artifact_name and len(self.artifact_name) != 32:
```

Comment:
Hmmm, can we do this another way, by checking if the digest exists? Seems kinda gnar to have this special case.

Reply:
This is already in a block where we know whatever was passed in doesn't exist or we don't have permissions to view.  This check only changes whether we raise a "Artifact without alias error" or an "Artifact does not exist error".

Reply:
Another option: we can make a one line change to gorilla to make it error if the name wasn't a digest or a valid alias. Would need to make sure that the CLI bubbles it up in a way that makes it obvious what went wrong.